wremnants/datasets/datagroups.py

```diff
@@ -220,13 +246,41 @@ def setHists(self, baseName, syst, procsToRead=None, label=None, nominalIfMissin
                         h = h.project(*projections)
 
                 if preOpMap and member.name in preOpMap:
-                    logger.debug(f"Applying preOp to {member.name} after loading")
+                    logger.debug(f"Applying preOp to {member.name}/{procName} after loading")
                     h = preOpMap[member.name](h, **preOpArgs)
-
+                    
                 if self.globalAction:
                     h = self.globalAction(h)
-
-                group.hists[label] = hh.addHists(h, group.hists[label], createNew=False) if group.hists[label] else h
+                    logger.debug("Applying global action")
+
+                hasPartialSumForFake = False
+                if hasFake and procName != nameFake:
+                    if member.name in fakesMembers:
+                        if member.name not in fakesMembersWithSyst:
+                            fakesMembersWithSyst.append(member.name)
+                        hasPartialSumForFake = True
+                        # apply the correct scale for fakes
+                        scaleProcForFake = self.groups[nameFake].scale(member)
+                        logger.debug(f"Summing hist {read_syst} for {member.name} to {nameFake} with scale = {scaleProcForFake}")
+                        hProcForFake = scaleProcForFake * h.copy()
```

Comment:
Can we even avoid this h.copy by writing in the next line:
```
histForFake = hh.addHists(hProcForFake, histForFake, scale1=scaleProcForFake, createNew=False) if histForFake else hProcForFake.copy()
```
and then in hh.addHists we add the optional argument to scale the first histogram?